### DISCUSSION TITLE
Turborepo@2 is no longer playing well with vscode debug terminal

### DISCUSSION THREAD

**Discussion by notaphplover** [vercel/turborepo#8388] [Help]
### Summary

I'm not sure about creating an issue about this, but `turborepo@2` no longer play well with vscode debug terminal. Running a task in `turborepo@1` (including `turborepo@1.13`) in a vscode debug terminal triggers vscode debug breakpoints, but this behavior no longer happens when trying `turborepo@2`.

We can still use more traditional ways to debug a node application (the inspector protocol works like a charm), but the vscode debug terminal is such an straightforward approach I think it would be great to support it in turborepo.

Is this behavior expected?

Consider [this sample repo](https://github.com/notaphplover/turborepo-trap-issue) as an example. the task `foo:turbo` cannot be debugged using a javascript debug terminal unless `turbo@1` is installed. Setting `{ "ui": "stream" }` in the `turbo.json` config file does not solve the issue.

### Additional information

I wonder if `turborepo@2` spawns child task in a different way `turborepo@1` does. Maybe https://github.com/microsoft/vscode-js-debug/issues/1101#issuecomment-915345492 is relevant.

### Example

https://github.com/notaphplover/turborepo-trap-issue

**Comment by diogo1z** [comment]
I'm having the same issue =(

**Comment by anthonyshew** [comment]
Converting to an issue!

**Comment by netpoe** [comment]
I second this.
I was debugging my NodeJS apps with VSCode successfully until I added Turbo Repo `"turbo": "^2.2.3"`, my launch.json file changed from this to this:

```
  "version": "0.2.0",
  "configurations": [
    {
      "name": "Next.js: debug server-side",
      "type": "node-terminal",
      "request": "launch",
      "cwd": "${workspaceRoot}/apps/web",
      "envFile": "${workspaceFolder}/apps/web/.env",
      "command": "yarn run dev"
    },
```

```
"configurations": [
    {
      "name": "Next.js: debug server-side",
      "type": "node-terminal",
      "request": "launch",
      "cwd": "${workspaceRoot}",
      "envFile": "${workspaceFolder}/.env",
      "command": "yarn run dev"
    },
```

Launching the debugger with the original config throws this now:

```
yarn run dev
Debugger attached.
npm error code ENOWORKSPACES
npm error This command does not support workspaces.
npm error A complete log of this run can be found in: ~/.npm/_logs/2024-11-14T17_30_04_209Z-debug-0.log
Waiting for the debugger to disconnect...
nvm is not compatible with the npm config "prefix" option: currently set to ""
Run `nvm use --delete-prefix v20.9.0 --silent` to unset it.
```

So the updated launch.json config now launches the debugger but the breakpoints won't work.
